### DISCUSSION TITLE
Greenfield API: Endpoint for generating Addresses from Derivation Scheme

### DISCUSSION THREAD

**Discussion by eza** [btcpayserver/btcpayserver#3905] [Technical Support]
Would be handy to have an end point where I can post my derivation scheme (xpub and so on), select the type of Address (P2SH, ..) and I get the list of addresses as a response. Perhaps the address types should also have an end point where I can look up for them and send their ID on the main endpoint.

This would be useful when you have an app with users. The app is integrated with BTCPay Server and you want your users to automatically receive funds to their addresses rather than being custodial. Of course The whole address generation could be done on the app but since it's already available in BTCPay Server would be just handy to expose it to an API end point.

**Comment by bolatovumar** [comment]
I'm a bit torn on this one. Obviously, we could implement this, but it doesn't seem like this issue has anything to do with BTCPay specifically as it's just a generic "give me a list of addresses for this xpub" thing. @NicolasDorier any thoughts on this?
